### PR TITLE
TextEvent Fix for Windows/UWP

### PR DESCRIPTION
**Pull request by tomspilman**
This is the Windows DX and Windows UWP support for the change in https://github.com/mono/MonoGame/pull/4878.

### CODE REVIEW

**Comment by tomspilman** [comment]
@dellis1972 @KonajuGames 

This basically matches your change for SDL2 to the other platforms that support this event.  Still has the same flaws as the other.

**Comment by tomspilman** [comment]
Looks like I need to fix the old MonoMac code path.

@dellis1972 @cra0zy - Were we going to remove that in 3.7?

File: MonoGame.Framework/TextInputEventArgs.cs [hunk 1]
```diff
@@ -8,25 +8,30 @@
...
 {
     /// <summary>
-    /// This class is used for the game window's TextInput event as EventArgs.
+    /// An event containing keyboard text input data.  See
+    /// <see cref="GameWindow.TextEvent"/> for details.
     /// </summary>
     public class TextInputEventArgs : EventArgs
     {
-        char character;
-        public TextInputEventArgs(char character, Keys key = Keys.None)
```
Comment (dellis1972) [inline MonoGame.Framework/TextInputEventArgs.cs]: @tomspilman put the default back in for "key" and the monomac build will pass.

**Comment by dellis1972** [comment]
@tomspilman I want to see MonoMac gone asap, its totally redundant now that Xamarin.Mac is free.

File: MonoGame.Framework/TextInputEventArgs.cs (hunk 1 above)
Comment (tomspilman) [inline MonoGame.Framework/TextInputEventArgs.cs]: I am aware of that...  I didn't want to let it pass that way.  I wanted to fix the other platforms that use the event to have matching behavior to your changes in DesktopGL.

**Comment by tomspilman** [comment]
> I want to see MonoMac gone asap, its totally redundant now that Xamarin.Mac is free.

It definitely has some stinky bits... like how it re-implements `GameWindow` instead of inheriting from it.

I would like the remove redundant platforms... MonoMac, Linux... now what we have SDL2.  Still i worry about breaking `Protobuild -generate MacOS` and `Protobuild -generate Linux` and how it would affect some users.

**Comment by dellis1972** [comment]
@tomspilman can you rebase this PR?

**Comment by tomspilman** [comment]
@dellis1972 - This cannot be fixed as is.  

After i wrote this PR we merged a bunch of other changes to Windows text input over it which broke the approach.

I'm going to close this for now and put in an issue to revisit it fresh.  See https://github.com/MonoGame/MonoGame/issues/6204

**Pull request closed without merging**